helix-view/src/document.rs

```diff
@@ -1549,7 +1549,7 @@ impl Document {
     pub fn relative_path(&self) -> Option<PathBuf> {
```

Comment:
For this case I think we should either return an `Option<Result<PathBuf>>` or keep the current `Option<PathBuf>` return and return the absolute path when `get_relative_path` fails (and document that behavior in the function docs here).

Or I suppose we could remove this function and have the callers use `Document::path` and handle getting the relative path themselves since the `Option<Result<PathBuf>>` type is so awkward.

Reply:
I went for the second option -> keeping the current `Option<PathBuf>` but returning the absolute path as a fallback. It feel more coherent with other places where we apply the same logic :)